File: backend/src/Assignment.Domain/Entities/Assignment.cs
namespace Assignment.Domain.Entities;

public class Assignment
{
    public Guid Id { get; set; }

    public Guid CourseId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public DateTime DueDate { get; set; }

    public decimal TotalMarks { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

}

File: frontend/types/assignment.ts
export interface Assignment {
  id: string;
  courseId: string;
  title: string;
  description: string;
  dueDate: string;
  totalMarks: number;
  createdAt: string;
}

export interface AssignmentRequest {
  courseId: string;
  title: string;
  description: string;
  dueDate: string;
  totalMarks: number;
}
